src/matrix_t/matrix_t.h

```diff
@@ -77,7 +103,8 @@ class matrix_t {
       ch = channel;
     }
     // relocate buffer only if new size doesnt fit
-    int new_size = ((c * get_data_type_size(type) * ch) * r);
+    //int new_size = ((c * get_data_type_size(type) * ch) * r);
+    int new_size = (cols * channel) * rows;
```

Comment:
this is not correct, it should be instead:
`int new_size = (c * ch) * r;`
because it should take in account the values from the arguments not from properties. Otherwise we can't know the `new_size`...